lib/codegen/fromcto/rust/rustvisitor.js

```diff
@@ -225,6 +233,39 @@ class RustVisitor {
             }
         }
         parameters.fileWriter.writeLine(1, ')]');
+
+        if (!ModelUtil.isPrimitiveType(field.type) && ModelUtil.isMap(field)) {
+            const mapDeclaration = field.getModelFile().getType(field.getType());
+            const mapKeyType     = mapDeclaration.getKey().getType();
+            const mapValueType   = mapDeclaration.getValue().getType();
+
+            let rustKeyType;
+            let rustValueType;
+
+            // Key
+            if (ModelUtil.isPrimitiveType(mapKeyType)) {
+                rustKeyType = this.toRustType(mapKeyType);
+            } else if (ModelUtil.isScalar(mapDeclaration.getKey())) {
+                const scalarDeclaration = mapDeclaration.getModelFile().getType(mapDeclaration.getKey().getType());
+                const scalarType = scalarDeclaration.getType();
+                rustKeyType = this.toRustType(scalarType);
```

Comment:
`toRustType` definition is lacking `Integer` conversion, here https://github.com/accordproject/concerto-codegen/blob/41428eda6da56ffb9fe102a90eccc01d6757916f/lib/codegen/fromcto/rust/rustvisitor.js#L289-L303

`Integer` is one of the possibilities, here https://github.com/accordproject/concerto/blob/5b72c1951c4f1546cbc29a8a18e89d516efa74e3/packages/concerto-core/lib/introspect/scalardeclaration.js#L61-L75

I think we can convert `Integer` to `u32`.

Reply:
Also checked `DataTime<Utc>` from the `chronos` library, it is hashable. So we are safe there.

Reply:
JavaScript uses double-precision floating-point format, which is a 64-bit binary format defined by the IEEE 754-2008 standard. I think `f64` is better suited?

Reply:
It seems counter-intuitive, my line of thinking is there will be no loss of precision when working with JS numbers

Reply:
I have two counterpoints for using `f64`. First: I would be very surprised if I see a float number when my concerto type is an integer. Especially the point of code generation is not JS compatibility. 

More importantly, the second issue is that `f64` does not implement hash. So it would break for keys.

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=e6f21b9a4a19946b348a3ac81f93293a

Reply:
I guess the 3rd one is that for `Long` we are converting them to u64, which should have been u128 anyway, but for integers it could at least be `i64`. I don't know, maybe @martinhalford can weight in.

Reply:
Concerto documentation dictates that Integers should be `i32` and longs `i64` https://concerto.accordproject.org/docs/design/specification/model-properties#:~:text=Integer,64%20bit%20signed%20whole%20number. so existing conversion of `Long` is also wrong?